### PR TITLE
cut release 0.11

### PR DESCRIPTION
**Pull request by raptorsun**
## Description

_Describe the big picture of your changes here to communicate to the maintainers why we should accept this pull request.
If it fixes a bug or resolves a feature request, be sure to link to that issue._

This PR prepares for cutting release 0.11:
- Updated changelog
- Updated compatibility matrix
- Pinned dependencies

## Type of change

_What type of changes does your code introduce to the kube-prometheus? Put an `x` in the box that apply._

- [ ] `CHANGE` (fix or feature that would cause existing functionality to not work as expected)
- [ ] `FEATURE` (non-breaking change which adds functionality)
- [ ] `BUGFIX` (non-breaking change which fixes an issue)
- [ ] `ENHANCEMENT` (non-breaking change which improves existing functionality)
- [x] `NONE` (if none of the other choices apply. Example, tooling, build system, CI, docs, etc.)

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -38,12 +38,12 @@ This adapter is an Extension API Server and Kubernetes needs to be have this fea
...
+| kube-prometheus stack                                                                      | Kubernetes 1.20 | Kubernetes 1.21 | Kubernetes 1.22 | Kubernetes 1.23 | Kubernetes 1.24 |
 |--------------------------------------------------------------------------------------------|-----------------|-----------------|-----------------|-----------------|-----------------|
-| [`release-0.7`](https://github.com/prometheus-operator/kube-prometheus/tree/release-0.7)   | ✔               | ✔               | ✗               | ✗               | ✗               |
-| [`release-0.8`](https://github.com/prometheus-operator/kube-prometheus/tree/release-0.8)   | ✗               | ✔               | ✔               | ✗               | ✗               |
-| [`release-0.9`](https://github.com/prometheus-operator/kube-prometheus/tree/release-0.9)   | ✗               | ✗               | ✔               | ✔               | ✗               |
-| [`release-0.10`](https://github.com/prometheus-operator/kube-prometheus/tree/release-0.10) | ✗               | ✗               | ✗               | ✔               | ✔               |
```
Comment (paulfantom) [inline README.md:46]: We need to change https://github.com/prometheus-operator/kube-prometheus/blob/main/.github/workflows/ci.yaml#L86-L87 to reflect that we are testing on kubernets 1.24.

Comment (raptorsun) [reply README.md:46]: Thanks, I will put 1.24 image to that list :)

**Comment by paulfantom** [comment]
@raptorsun could you run `make docs` to fix docs formatting issues reported by CI?

**Comment by raptorsun** [comment]
> @raptorsun could you run `make docs` to fix docs formatting issues reported by CI? Also it seems there are issues in e2e tests that need investigation before we merge this.

The document formatting error is the link to the release branch `https://github.com/prometheus-operator/kube-prometheus/tree/release-0.11` is not yet valid. I'm going to remove that link and put it back in the follow-up pull request. 

I'm checking the failed E2E test at the moment, will make a fix once the cause is found.

**Comment by paulfantom** [comment]
> The document formatting error is the link to the release branch https://github.com/prometheus-operator/kube-prometheus/tree/release-0.11 is not yet valid

Ah, sorry, I didn't check this. Let's leave it in then. This CI check is not required to merge the PR.

**Comment by raptorsun** [comment]
> > The document formatting error is the link to the release branch https://github.com/prometheus-operator/kube-prometheus/tree/release-0.11 is not yet valid
> 
> Ah, sorry, I didn't check this. Let's leave it in then. This CI check is not required to merge the PR.

No problem, the link is kept in the new commit. 
The issue on E2E test for k8s 1.24 has been fixed by upgrading kind from v0.12 to v0.14.

**Comment by paulfantom** [comment]
Release is done. @raptorsun could you unlock `main` branch by removing pinning to particular branches in `jsonnet/kube-prometheus/jsonnetfile.json`?

**Comment by raptorsun** [comment]
The follow-up PR for release 0.11 is ready:
https://github.com/prometheus-operator/kube-prometheus/pull/1786

**Pull request merged**